app/routes/errors.py

```diff
@@ -121,47 +145,54 @@ def submission_failed(error=None):
     return _render_error_page(500, template="submission-failed")
 
 
-@errors_blueprint.app_errorhandler(Exception)
-def internal_server_error(error=None):
-    try:
-        log_error(error, 500)
-        return _render_error_page(500)
-    except Exception:  # pylint:disable=broad-except
-        logger.error(
-            "an error has occurred when rendering 500 error",
-            exc_info=True,
-            url=request.url,
-            status_code=500,
-        )
-        return render_template(template="errors/500"), 500
-
-
-@errors_blueprint.app_errorhandler(403)
-@errors_blueprint.app_errorhandler(404)
-def http_exception(error):
-    log_error(error, error.code)
-    return _render_error_page(error.code)
-
-
-@errors_blueprint.app_errorhandler(FulfilmentRequestFailedException)
-def fulfilment_request_failed(_):
-    logger.exception(
-        "An individual response fulfilment request failed",
-        url=request.url,
-        status_code=500,
-    )
+@errors_blueprint.app_errorhandler(FulfilmentRequestPublicationFailed)
+def fulfilment_request_publication_failed(error):
+    log_error(error, 500)
 
-    if "mobile_number" in request.args:
-        blueprint_method = (
-            "individual_response.individual_response_text_message_confirm"
+    if isinstance(error.invoked_by, ConfirmationEmail):
```

Comment:
I'm really not a fan of using isinstance for logic (it's too easy to get it wrong down the line via inheritance), but I don't see a better way to do this using the current approach of a single FulfilmentRequestPublicationFailed exception. My other argument against using the `isinstance` approach is that the relevant error handler function is now doing multiple things. I'd prefer to use specific errors in this case

Reply:
This was in an attempt to not have really long names for exceptions but I have now used explicit exceptions for each case to make it easier to digest 👍